### PR TITLE
Compute 2022-07-02, 2022-08-01 changes 

### PR DESCRIPTION
**Pull request by haagha**
# Contributing to the Azure SDK
This PR includes changes from 
DiskRP version 2022-07-02: https://github.com/Azure/azure-rest-api-specs/pull/20223
and ComputeRP versions 2022-08-01 https://github.com/Azure/azure-rest-api-specs/pull/20552

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by Yao725 — DISMISSED** [review]
Where does these code changes come from? I don't see any metadata change [here](https://github.com/Azure/azure-sdk-for-net/blob/main/eng/mgmt/mgmtmetadata/compute_resource-manager.txt).

And one more important thing is, this is the track 1 SDK and we will deprecate it soon, please contact @ArthurMa1978 and provide the justification about this release.

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations/SupportedCapabilities.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Disk update resource.
+    /// </summary>
+    public partial class SupportedCapabilities
+    {
+
+        public SupportedCapabilities(bool? acceleratedNetwork = default(bool?), string architecture = default(string))
```
Comment (grizzlytheodore) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations/SupportedCapabilities.cs:13]: here acceleratedNetwork should not have default

**Comment by haagha** [comment]
For compute we discussed with @ArthurMa1978 that track2 will be released.

**Comment by haagha** [comment]
New Version increment is not needed for this PR now. @grizzlytheodore will add his changes and make changes for new release.

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj [hunk 2]
```diff
@@ -9,7 +9,7 @@
       Provides developers with libraries for the updated compute platform under Azure Resource manager to deploy virtual machine, virtual machine extensions and availability set management capabilities. Launch, restart, scale, capture and manage VMs, VM Extensions and more. Note: This client library is for Virtual Machines under Azure Resource Manager.
       Development of this library has shifted focus to the Azure Unified SDK. The future development will be focused on "Azure.ResourceManager.Compute" (https://www.nuget.org/packages/Azure.ResourceManager.Compute/). Please see the package changelog for more information.
     </Description>
-    <Version>57.0.0</Version>
+    <Version>58.0.0</Version>
```
Comment (ArthurMa1978) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj:12]: please also update the AzSdk.RP.props file

File: sdk/compute/Microsoft.Azure.Management.Compute/tests/SessionRecords/DiskRPScenarioTests/Disk_CRUD_WithSecurityProfile_Import.json [hunk 3]
```diff
@@ -2748,20 +2689,20 @@
           "-1"
         ]
       },
-      "ResponseBody": "{\r\n  \"startTime\": \"2022-06-15T01:16:50.0290755-04:00\",\r\n  \"endTime\": \"2022-06-15T01:16:50.2321632-04:00\",\r\n  \"status\": \"Succeeded\",\r\n  \"properties\": {\r\n    \"output\": {\r\n      \"accessSAS\": \"Sanitized\",\r\n      \"securityDataAccessSAS\": \"{SAS}\"\r\n    }\r\n  },\r\n  \"name\": \"4f08a815-6459-4d9d-aae9-947e11c289dc\"\r\n}",
+      "ResponseBody": "{\r\n  \"startTime\": \"2022-08-19T10:46:36.5488408-04:00\",\r\n  \"endTime\": \"2022-08-19T10:46:36.7207032-04:00\",\r\n  \"status\": \"Succeeded\",\r\n  \"properties\": {\r\n    \"output\": {\r\n      \"accessSAS\": \"Sanitized\",\r\n      \"securityDataAccessSAS\": \"https://md-hdd-3glxsvzdqvc4.z33.blob.storage.azure.net/zdsslsgqrg24/ca2e5663-d7c5-4d29-b6ac-2f17635b1cc8_vmgs?sv=2018-03-28&sr=b&si=55143673-16cd-4be3-a9cd-ae72062f6eda&sig=e554pfmV1yEd%2FikwSUtu1u2kmnLCNqhAG99bWSktjBc%3D\"\r\n    }\r\n  },\r\n  \"name\": \"3998d449-c144-45b0-adde-2ca5ebcf8376\"\r\n}",
```
Comment (ArthurMa1978) [inline sdk/compute/Microsoft.Azure.Management.Compute/tests/SessionRecords/DiskRPScenarioTests/Disk_CRUD_WithSecurityProfile_Import.json:2692]: Please follow this https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.Management.ServiceBus/tests/Tests/ScenarioTests.cs#L34-L36 sample to sanitize all sas tokens

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/Disk.cs [hunk 4]
```diff
@@ -130,7 +132,14 @@ public Disk()
...
...SizeGB = default(int?), long? diskSizeBytes = default(long?), string uniqueId = default(string), EncryptionSettingsCollection encryptionSettingsCollection = default(EncryptionSettingsCollection), string provisioningState = default(string), long? diskIOPSReadWrite = default(long?), long? diskMBpsReadWrite = default(long?), long? diskIOPSReadOnly = default(long?), long? diskMBpsReadOnly = default(long?), string diskState = default(string), Encryption encryption = default(Encryption), int? maxShares = default(int?), IList<ShareInfoElement> shareInfo = default(IList<ShareInfoElement>), string networkAccessPolicy = default(string), string diskAccessId = default(string), System.DateTime? burstingEnabledTime = default(System.DateTime?), string tier = default(string), bool? burstingEnabled = default(bool?), PropertyUpdatesInProgress propertyUpdatesInProgress = default(PropertyUpdatesInProgress), bool? supportsHibernation = default(bool?), DiskSecurityProfile securityProfile = default(DiskSecurityProfile), double? completionPercent = default(double?), string publicNetworkAccess = default(string), string dataAccessAuthMode = default(string), bool? optimizedForFrequentAttach = default(bool?))
```
Comment (grizzlytheodore) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/Disk.cs:142]: need customization

**Comment by grizzlytheodore** [comment]
this is ready for merge @ArthurMa1978 @Yao725

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**